File: backend-python/models/booking.py
from typing import Optional, TYPE_CHECKING
from sqlmodel import SQLModel, Field, Relationship  # type: ignore

if TYPE_CHECKING:
    from .user import User
    from .resModel import Restaurant


class Booking(SQLModel, table=True):
    bookingId: Optional[int] = Field(default=None, primary_key=True)
    userId: int = Field(foreign_key="user.userId")
    restaurantId: int = Field(foreign_key="restaurants.id")
    date: str
    time: str
    guestCount: int
    requestSeats: int
    assignedSeats: int = Field(default=0)
    status: str = Field(default="pending")
    contactName: str
    contactEmail: str
    contactPhone: str
    note: Optional[str] = None
    createdAt: Optional[str] = None
    isPaid: bool = Field(default=False)

    user: Optional["User"] = Relationship()
    restaurant: Optional["Restaurant"] = Relationship()

File: frontend-react/src/types/booking.ts
export interface BookingMenuItemInput {
  itemId: number;
  quantity: number;
}

export interface BookingMenuItemDetail {
  bookingItemId: number;
  itemId: number;
  quantity: number;
  price: number;
  name: string;
  category: string;
  image_url?: string | null;
  description?: string | null;
}

export interface BookingDetail {
  bookingId: number;
  userId: number;
  restaurantId: number;
  restaurantName?: string | null;
  date: string;
  time: string;
  guestCount: number;
  childCount: number;
  requestSeats: number;
  assignedSeats: number;
  status: string;
  cancellationStatus?: string | null;
  cancellationReason?: string | null;
  cancellationEvidence?: string | null;
  cancellationActor?: string | null;
  depositAmount: number;
  depositStatus: string;
  refundStatus?: "pending" | "processing" | "refunded" | null;
  restaurantReportStatus?: "open" | "appeal_pending" | "dismissed" | "appeal_rejected" | null;
  depositPaidAt?: string | null;
  depositExpiresAt?: string | null;
  contactName: string;
  contactEmail: string;
  contactPhone: string;
  note?: string | null;
  createdAt?: string | null;
  booking_items: BookingMenuItemDetail[];
}

export interface BookingCreatePayload {
  restaurantId: number;
  date: string;
  time: string;
  guestCount: number;
  childCount: number;
  requestSeats: number;
  contactName: string;
  contactEmail: string;
  contactPhone: string;
  note?: string;
  items: BookingMenuItemInput[];
}
